### PR TITLE
Allow to install untrusted packages on OSX

### LINKED ISSUE #117: Failed to install CLTools on OSX 10.7
### Cookbook version
6.0.3

### Chef-client version
Chef Development Kit Version: 0.15.16
chef-client version: 12.11.18
berks version: 4.3.5
kitchen version: 1.10.0

### Platform Details
OSX 10.7

### Scenario:
kitchen converge

### Steps to Reproduce:
You need an OSX 10.7 box. Add
```
  - name: osx-10.7
    driver:
      box: osx-10.7
      box_url: url to OSX 10.7 box
```
in .kitchen.yml platforms section and run `kitchen converge default-osx-107`.

### Expected Result:
I'm expecting CLTools package installed.

### Actual Result:
```
  * execute[installer -package "$(find '/tmp/kitchen/cache/osx-command-line-tools' -name *.mpkg)" -target "/"] action run
             
             ========================================================================
             Error executing action `run` on resource 'execute[installer -package "$(find '/tmp/kitchen/cache/osx-command-line-tools' -name *.mpkg)" -target "/"]'
             ========================================================================
             
             Mixlib::ShellOut::ShellCommandFailed
             ------------------------------------
             Expected process to exit with [0], but received '1'
             ---- Begin output of installer -package "$(find '/tmp/kitchen/cache/osx-command-line-tools' -name *.mpkg)" -target "/" ----
             STDOUT: installer: Package name is Command Line Tools (Lion)
             installer: Certificate used to sign package is not trusted. Use -allowUntrusted to override.
             STDERR: 
             ---- End output of installer -package "$(find '/tmp/kitchen/cache/osx-command-line-tools' -name *.mpkg)" -target "/" ----
             Ran installer -package "$(find '/tmp/kitchen/cache/osx-command-line-tools' -name *.mpkg)" -target "/" returned 1
```

### PR DESCRIPTION
**Pull request by atikhono**
### Description

Fixes failing converge on OSX 10.7

### Issues Resolved

#117 

### Check List
- [x] All tests pass. See https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD
- [ ] New functionality includes testing.
N/A Makes it work at all
- [ ] New functionality has been documented in the README if applicable
N/A
- [x] The CLA has been signed. See https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD

### CODE REVIEW

File: libraries/xcode_command_line_tools.rb [hunk 1]
```diff
@@ -154,7 +154,7 @@ def attach
     # @return [void]
     #
     def install
-      execute %|installer -package "$(find '#{mount_path}' -name *.mpkg)" -target "/"|
+      execute %|installer -allowUntrusted -package "$(find '#{mount_path}' -name *.mpkg)" -target "/"|
```
Comment (schisamo) [inline libraries/xcode_command_line_tools.rb]: The `-allowUntrusted` flag should only be included with the affected platform version so `10.7` in this case. We want the default behavior to only allow trusted packages to be installed. You'll also want to add a big comment explaining why the `-allowUntrusted` flag is being used on these versions...maybe something like:

```ruby
# The "-allowUntrusted" flag has been added to the installer
# command to accommodate for now-expired certificates used
# to sign the downloaded command line tools.
```

Comment (atikhono) [inline libraries/xcode_command_line_tools.rb]: This is executed only on OSX 10.7 and 10.8 (when CLTools are installed from dmg). Unfortunately, I don't have an OSX 10.8 box to check if it needs -allowUtrusted flag. Maybe we can leave the code as is and just add the comment?

Comment (schisamo) [inline libraries/xcode_command_line_tools.rb]: Let's definitely start by adding the comment. @tas50 @jtimberman Any thoughts?

**Comment by Kasen** [comment]
@tas50 @jtimberman, guys, please tell your opinion.

**Comment by tas50** [comment]
It would be nice to get a comment that explained why we're doing this. It's often hard years later to do the detective work to figure out why important flags like this were included.

File: libraries/xcode_command_line_tools.rb (hunk 1 above)
Comment (atikhono) [inline libraries/xcode_command_line_tools.rb]: @schisamo done

**Pull request merged**